azimuth/dataset_split_manager.py

```diff
@@ -326,10 +326,7 @@ def get_tags(
             ds = ds.select(indices)
         available_tags = self._tags if table_key is None else self._tags + self._prediction_tags
 
-        return {
-            row_idx: {tag: ds[tag][idx] for tag in available_tags}
-            for idx, row_idx in enumerate(ds["row_idx"])
-        }
+        return {row[DatasetColumn.row_idx]: {tag: row[tag] for tag in available_tags} for row in ds}
```

Comment:
Do you think that `ds.map()` could be faster than a `for` loop on `ds`? Something like this (not tested):
```suggestion
        return dict(
            ds.map(
                lambda row: (row[DatasetColumn.row_idx], {tag: row[tag] for tag in available_tags}),
                input_columns=[DatasetColumn.row_idx, *available_tags],
            )
        )
```

Reply:
I'll profile and get back to you!!

Reply:
In fact, I just realized that in the two places that we call `get_tags()`, we then do `.values()`, so I think we can avoid the `dict` altogether!
```suggestion
        ds.map(
            lambda row: {tag: row[tag] for tag in available_tags},
            input_columns=[DatasetColumn.row_idx, *available_tags],
        )
```

Reply:
The goal was to be consistent with the `add_tags()` method - might not be needed though.

Reply:
I see.

Reply:
I get `TypeError: <lambda>() takes 1 positional argument but 24 were given`. I'm not familiar enough with the `map() ` function to debug quickly.

Reply:
Oh, right! Forget my `input_columns`. Without it, it should work, I think.

Reply:
I am not familiar at all. I might be completely off. We can forget about it.